engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -73,6 +79,10 @@ public String apply(@Nullable Level input)
     }
   };
 
+  static {
```

Comment:
Agree. The implementation of MDC allows to set MDC properties for a thread.

But I disagree with your statement "It will not work for unit tests where both app master and containers/workers run in the same JVM but different threads." If somebody needs to implement the unit test that would run the master and containers in different threads, he or she can call the method setService() in the unit test code explicitly.